WeatherDataCalculator.py

```diff
@@ -0,0 +1,82 @@
+
+
+class WeatherDataCalculator:
+    def __init__(self, weather_data_calculations, weather_data):
+        self.weather_data_calculations = weather_data_calculations
+        self.weather_data = weather_data
+
+    def calculate_year_results(self, year):
+        if year in self.weather_data.keys():
+            max_temp, max_month, max_day, min_temp, min_month, min_day, max_humidity, max_humidity_month, max_humidity_day = \
+                self.calculate_temperatures(year)
+            self.weather_data_calculations[year] = {
+                "max_temp": max_temp,
+                "max_month": max_month,
+                "max_day": max_day,
+                "min_temp": min_temp,
+                "min_month": min_month,
+                "min_day": min_day,
+                "max_humidity": max_humidity,
+                "max_humidity_month": max_humidity_month,
+                "max_humidity_day": max_humidity_day
+            }
+
+    def calculate_temperatures(self, year):
+        max_temp = int(list(list(self.weather_data[year].items())[
+                       0][1].items())[0][1]["Max TemperatureC"])
+        max_month = list(self.weather_data[year].keys())[0]
+        max_day = list(list(self.weather_data[year].items())[0][1].keys())[0]
+        min_temp = int(list(list(self.weather_data[year].items())[
+                       0][1].items())[0][1]["Min TemperatureC"])
+        min_month = max_month
+        min_day = max_day
+        max_humidity = int(list(list(self.weather_data[year].items())[
+                           0][1].items())[0][1]["Max Humidity"])
```

Comment:
what are we doing here?

Reply:
`calculte_temperatures` calculates the Highest, Lowest Temperature and its Day and Month and Highest Humidity and its day and Month.... So in 
`max_temp = int(list(list(self.weather_data[year].items())[0][1].items())[0][1]["Max TemperatureC"])`
I am assigning initial value to `max_temp` and that initial value is the max temperature of the first day in the first month of that year.... and in `max_month` i am assigning the month of that initial value and son